drivers/sensor/bmi160/bmi160.h

```diff
@@ -296,8 +296,10 @@ enum bmi160_odr {
 #define BMI160_GYR_RANGE_125DPS		4
 
 #ifdef CONFIG_SENSOR_VERSION_2
-#define BMI160_ACC_SCALE(range_g)   fp_div(INT_TO_FP(range_g), INT_TO_FP(INT16_MAX))
-#define BMI160_GYR_SCALE(range_dps) fp_div(INT_TO_FP(range_dps), INT_TO_FP(INT16_MAX))
+#define BMI160_ACC_SCALE(range_g)                                                                  \
+	fp_div(fp_mul(INT_TO_FP(range_g), SENSOR_G), INT_TO_FP(INT16_MAX))
```

Comment:
I'm not totally following the math changes in this PR, but weren't these facilities already available in some form in the previous API? And doesn't the new API support floating point and fixed point with the same syntax? Is it possible to use the same calls regardless of the API in this driver?